tests/integration/netdev.bats

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	requires root
+	setup_busybox
+
+	# skip to setup resources that require root
+	# to avoid the test to fail
+	if [ $EUID -ne 0 ] || in_userns; then
+		# create a dummy interface to move to the container
+		ip link add dummy0 type dummy
+
+		# create a temporary name for the test network namespace
+		tmp=$(mktemp)
+		rm -f "$tmp"
+		ns_name=$(basename "$tmp")
+
+		# create network namespace
+		ip netns add "$ns_name"
+		ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+	fi
+}
+
+function teardown() {
+	if [ $EUID -ne 0 ] || in_userns; then
+		ip netns del "$ns_name"
+		ip link del dev dummy0
+	fi
+
+	teardown_bundle
+}
+
+@test "move network device to container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+      		| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to container network namespace and restore it back" {
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	update_config ' .linux.netDevices |= {"dummy0": {} }'
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	# the network namespace owner controls the lifecycle of the interface
+	ip netns exec "$ns_name" ip link set dev dummy0 netns 1
+	[ "$status" -eq 0 ]
+
+	runc delete --force test_busybox
+
+	# verify the interface is back in the root network namespace
+	ip address show dev dummy0
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to precreated container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+      		| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	# verify the interface is still present in the network namespace
+	ip netns exec "$ns_name" ip address show dev dummy0
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to precreated container network namespace and set ip address with global scope" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+      		| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	global_ip="169.254.169.77"
+
+	# set a custom address to the interface
+	# set the interface down to avoid network problems
+	ip link set down dev dummy0
+	ip address add "$global_ip" dev dummy0
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
```

Comment:
nit: can you move it close to other `update_config` (just for the sake of readability)?

Reply:
done